### PR TITLE
[Core] Separate `PointObject` in different file

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Separate `PointObject` in different file

**🆕 Changelog**

- [Separate PointObject in different file](https://github.com/KratosMultiphysics/Kratos/commit/c83025b5d3e8342c7690a4a498c2bf1a34e3ace1)

### CODE REVIEW

**Review by RiccardoRossi — DISMISSED** [review]

File: kratos/spatial_containers/point_object.h [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    void pSetObject(typename TObject::Pointer pObject)
+    {
+        mpObject = pObject;
+        UpdatePoint();
+    }
+
+    /**
+     * @brief This method checks everything is right
+     */
+    void Check()
```
Comment (philbucher) [inline kratos/spatial_containers/point_object.h]:
```suggestion
    void Check() const
```

File: kratos/spatial_containers/point_object.h [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    ///@}
+    ///@name Acess
+    ///@{
+
+    /**
+     * @brief Returns the geometry associated to the point
+     * @return mrGeometry The reference to the geometry associated to the point
+     */
+    typename TObject::Pointer pGetObject()
```
Comment (philbucher) [inline kratos/spatial_containers/point_object.h]:
```suggestion
    typename TObject::Pointer pGetObject() const
```

**Review by philbucher — CHANGES_REQUESTED** [review]
minor

File: kratos/spatial_containers/point_object.h (hunk 2 above)
Comment (loumalouomega) [reply kratos/spatial_containers/point_object.h]: Done

File: kratos/spatial_containers/point_object.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/spatial_containers/point_object.h]: Done

**Review by philbucher — APPROVED** [review]

**Comment by loumalouomega** [comment]
@philbucher the intel compiler is still failing

**Comment by philbucher** [comment]
I removed the legacy compiler (icc) from the docker image ...

I will try to fix it in the next days

**Pull request merged**